execution/executor/src/block_executor/mod.rs

```diff
@@ -40,21 +40,31 @@ use aptos_types::{
 use aptos_vm::AptosVM;
 use block_tree::BlockTree;
 use fail::fail_point;
-use std::{marker::PhantomData, sync::Arc};
+use std::sync::Arc;
 
 pub mod block_tree;
 
 pub trait TransactionBlockExecutor: Send + Sync {
+    fn new() -> Self;
+
     fn execute_transaction_block(
+        &self,
         transactions: ExecutableTransactions,
         state_view: CachedStateView,
         onchain_config: BlockExecutorConfigFromOnchain,
         append_state_checkpoint_to_block: Option<HashValue>,
     ) -> Result<ExecutionOutput>;
 }
 
-impl TransactionBlockExecutor for AptosVM {
+pub struct AptosVMBlockExecutor;
```

Comment:
I know it is clear, but please add documentation? E.g., "block executor used in production by Aptos blockchain ...", it is really hard to navigate through the zoo of undocumented structs which do same things and are called similarly